staging/src/k8s.io/client-go/rest/request_test.go

```diff
@@ -3911,3 +3912,50 @@ func TestRequestBodyResetOrder(t *testing.T) {
 		t.Errorf("Expected invocation request and response body operations for retry do not match: %s", cmp.Diff(expected, recorder.order))
 	}
 }
+
+func TestRequestConcurrencyWithRetry(t *testing.T) {
+	var attempts int32
+	client := clientForFunc(func(req *http.Request) (*http.Response, error) {
+		defer func() {
+			atomic.AddInt32(&attempts, 1)
+		}()
+
+		// always send a retry-after response
+		return &http.Response{
+			StatusCode: http.StatusInternalServerError,
+			Header:     http.Header{"Retry-After": []string{"1"}},
+		}, nil
+	})
+
+	req := &Request{
+		verb: "POST",
```

Comment:
https://github.com/kubernetes/kubernetes/blob/cecc563d3b9a9438cd3e6ae1576baa0a36f2d843/staging/src/k8s.io/client-go/rest/request.go#L844-L856

?

Reply:
retries are happening via `{429|5xx} Retry-After: N` response from the server, not from retryable `err`

The test verifies that retries were attempted as expected:

https://github.com/kubernetes/kubernetes/pull/112072/files#diff-c0689eb7d67ca7fcd631394535d609de047419dc422ecf21df0b48324651ef1cR3956-R3960

`isErrRetryableFunc` is not relevant here since the server here is returning a `response` and `err` is `nil`

Reply:
ahh right, so it is only not retriable on methods that modifies the state and we don't know the result ... got it